src/main/java/io/supertokens/multitenancy/Multitenancy.java

```diff
@@ -396,7 +396,7 @@ public static boolean addUserIdToTenant(Main main, TenantIdentifierWithStorage t
                 String tenantId = tenantIdentifierWithStorage.getTenantId();
                 AuthRecipeUserInfo userToAssociate = storage.getPrimaryUserById_Transaction(tenantIdentifierWithStorage.toAppIdentifier(), con, userId);
 
-                if (userToAssociate.isPrimaryUser) {
+                if (userToAssociate != null && userToAssociate.isPrimaryUser) {
                     Set<String> emails = new HashSet<>();
                     Set<String> phoneNumbers = new HashSet<>();
                     Set<LoginMethod.ThirdParty> thirdParties = new HashSet<>();
```

Comment:
the checks below for same email etc for account linking should check:
- if the login method is the same, throw duplicaremalexception etc..
- if it's a different login method, then we shuold throw a new exception type -> UserTenantAssociationNotAllowed with a reason string -> if this is on the backend SDK, then we should have the status reflected in there too

Reply:
Should the other places where this type of check also happens consider the login method being different?